### DISCUSSION TITLE
Is SteVe Suitable for Production Use?

### DISCUSSION THREAD

**Discussion by tanerg** [steve-community/steve#1583] [General]
Hi everyone,

I'm evaluating SteVe for a project and would like to know if it's suitable for a production environment. Specifically, I'm interested in hearing from those who have used it in a live setting.

How stable and reliable has it been in your experience?
Are there any major issues or limitations I should be aware of?
What kind of infrastructure do you recommend for running SteVe in production?
Any insights or advice would be greatly appreciated!

Thanks in advance!

**Comment by rhmltrbch** [comment]
That makes two of us who are interested in this topic. I can hardly find any content or documentation that provides clarity on it.
Topics like high availability and load balancing seem to be mentioned only in passing within issues at best.

Have you made any progress with your inquiry?

**Reply by tanerg** [reply]
Unfortunately, I haven't heard from others yet, and my internet searches haven't turned up anything either. My own feeling is that SteVe is a good starting point, though I'm unsure if it would hold up in a large-scale environment.

**Comment by juherr** [comment]
I think https://github.com/steve-community/steve/issues/1643 is part of the answer.

Some CPO are using a Steve-based solution for their production.
The most visible one is https://github.com/parklapp/steve-pluggable

**Comment by goekay** [comment]
since i am partial to this topic (disclaimer: creator of steve), i did not reply and was waiting for others to chime in.

steve is a monolith: one java app and one database behind it. this will vertically scale only up to some degree. at some point you would need to split it into microservices across the critical zones. i cannot give you a number for the performance ceiling of this monolith though, since it depends on multiple criteria.

functionally, from a correctness perspective, steve _should_ be pretty solid. with this i mean: the things it implements should be correct. these were tested and validated during the last 10 years. if something does not work, people open an issue.

from a completeness perspective, there are some things missing in steve (e.g. _real_ dynamic smart charging, basic auth for websocket connections).

the HA and load balancing aspects depend on how you deploy and operate steve. steve, at the end of the day, is just the application. the ecosystem in which it exists matters. you could put proxies (nginx et al) in front of it, such that steve instance(s) are behind it running and processing messages.

----

this is only anecdotal but here we go: i receive feedback from time to time from commercial users of steve. some maintain 100s of charging stations. some companies are using it. some city transportation authorities are using it for their bus fleet.

**Reply by therealcmj** [reply]
Thanks for replying.

I see that Parlk forked it and made some, what appear on the surface to be, sensible changes. Is there any plan to merge their work back in?

**Reply by juherr** [reply]
@therealcmj Part of the answer will be there: https://github.com/steve-community/steve/issues/1733#issuecomment-2763376524
